tests/test_segmentclient.py

```diff
@@ -0,0 +1,545 @@
+        # todo: match the input and output label
+
+        gas_client.delete_dataset(dataset_name)
+
+    def test_delete_data(self, accesskey, url, tmp_path):
+        gas_client = GAS(access_key=accesskey, url=url)
+        dataset_name = get_random_dataset_name()
+        dataset_client = gas_client.create_dataset(dataset_name)
+        dataset_client.create_draft("draft-1")
+        dataset_client.upload_catalog(Catalog.loads(CATALOG))
+        segment_client = dataset_client.get_or_create_segment("segment1")
+
+        path = tmp_path / "sub"
+        path.mkdir()
+        for i in range(10):
+            local_path = path / f"hello{i}.txt"
+            local_path.write_text("CONTENT")
+            data = Data(local_path=str(local_path))
+            data.label = Label.loads(LABEL)
+            segment_client.upload_data(data)
+
+        segment_client.delete_data("hello0.txt")
+        data_paths = list(segment_client.list_data_paths())
+        assert "hello0.txt" not in data_paths
+
+        segment_client.delete_data(segment_client.list_data_paths())
+        data = list(segment_client.list_data())
+        assert len(data) == 0
+
+        gas_client.delete_dataset(dataset_name)
+
+    def test_sensor(self, accesskey, url):
+        gas_client = GAS(access_key=accesskey, url=url)
+        dataset_name = get_random_dataset_name()
+        dataset_client = gas_client.create_dataset(dataset_name, is_fusion=True)
+        dataset_client.create_draft("draft-1")
+        segment_client = dataset_client.get_or_create_segment("segment1")
+
+        segment_client.upload_sensor(Sensor.loads(LIDAR_DATA))
+
+        sensors = list(segment_client.list_sensors())
+        assert sensors
+        # todo: match the input and output sensors
+
+        segment_client.delete_sensor(LIDAR_DATA["name"])
+        sensors = list(segment_client.list_sensors())
+        assert len(sensors) == 0
+
+        gas_client.delete_dataset(dataset_name)
+
+    def test_upload_frame_without_label(self, accesskey, url, tmp_path):
+        gas_client = GAS(access_key=accesskey, url=url)
+        dataset_name = get_random_dataset_name()
+        dataset_client = gas_client.create_dataset(dataset_name, is_fusion=True)
+        dataset_client.create_draft("draft-1")
+        segment_client = dataset_client.get_or_create_segment("segment1")
+
+        segment_client.upload_sensor(Sensor.loads(LIDAR_DATA))
+
+        path = tmp_path / "sub"
+        path.mkdir()
+        for i in range(5):
+            frame = Frame()
+            local_path = path / f"hello{i}.txt"
+            local_path.write_text("CONTENT")
+            data = Data(local_path=str(local_path))
+            frame[LIDAR_DATA["name"]] = data
+            segment_client.upload_frame(frame, timestamp=i)
+
+        frames = list(segment_client.list_frames())
+        assert len(frames) == 5
+        assert frames[0][LIDAR_DATA["name"]].path == "hello0.txt"
+        assert not frames[0][LIDAR_DATA["name"]].label
```

Comment:
`# todo: match the input and output label`

Reply:
Because we assert that data has no label, there is no need to match the input and output label.